portality/article.py

```diff
@@ -156,10 +156,12 @@ def discover_duplicates(article, owner=None):
         if len(dois) > 0:
             # there should only be the one
             doi = dois[0]
-            articles = models.Article.duplicates(issns=issns, doi=doi)
-            possible_articles['doi'] = [a for a in articles if a.id != article.id]
-            if possible_articles['doi']:
-                found = True
+            # Only perform de-duplication on genuine DOIs - todo: do we need a full regex check?
+            if isinstance(doi, str) and doi.startswith('10.'):
+                articles = models.Article.duplicates(issns=issns, doi=doi)
+                possible_articles['doi'] = [a for a in articles if a.id != article.id]
+                if possible_articles['doi']:
+                    found = True
```

Comment:
I'm generally against testing for truthiness.  I'd do if len(possible_articles['doi']) > 0: found  = True